### PR TITLE
fix '-u' suggestion and snapshot summary

### PR DESCRIPTION
**Pull request by thomasbertet**
Hi,

# Goal 
- Have the "snapshot summary" with correct informations about added/unmatched/updated number of snapshots.
- Have the "u" & "i" usage suggestions in Jest watch mode.

# How
- I just sum-up the common counter with each value of the "child" state counters.

# Results
- Jest now displays the number of updated/added/unmatched snapshots + hints about the "u" to update snapshots.

# How to test
- Launch your example, add a story on the fly, Jest should display "1 snapshot written". 
- Edit a test, change the rendering, Jest should display `› 1 snapshot test failed in 1 test suite. Inspect your code changes or press `u` to update them.`

Let me know if you need anything else to have this merge :D 
Also, thanks for your work on this, I [believe there was some resistance](https://github.com/facebook/jest/issues/2676#issuecomment-274647517) from Jest core team to add this functionnality into core, so this package helps a lot :) 
Also you may want to follow what's [going on there](https://github.com/facebook/jest/pull/6143/files) since there might have a future official and less hacky way of doing it!

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,13 +20,23 @@ afterAll(() => {
...
+    }
   });
 });
 
 function toMatchSpecificSnapshot(received, snapshotFile, testName) {
   const absoluteSnapshotFile = getAbsolutePathToSnapshot(this.testPath, snapshotFile);
 
-  const commonSnapshotState = this.snapshotState;
+  // store the common state to re-use it in "afterAll" hook.
+  commonSnapshotState = this.snapshotState;
```
Comment (igor-dv) [inline src/index.js:39]: As I understand this will be replaced on every `toMatchSpecificSnapshot` call ?

Comment (thomasbertet) [reply src/index.js:39]: Indeed its is the only way we can get the original snapshotState. Not sure it worth it to add some checks to only store it the first time we get there. Wdty ?

Comment (igor-dv) [reply src/index.js:39]: I have to debug it a bit to understand the whole picture =)

Comment (thomasbertet) [reply src/index.js:39]: If you need more intel, please tell me :) Don't want to rush things, just would love to see this happen so it eases the process of updating snapshots in storyshots.

Comment (igor-dv) [reply src/index.js:39]: Sorry, will get to it this week. Just was a bit out of focus.

**Review by igor-dv — COMMENTED** [review]
@thomasbertet I've checked it now, and it looks working. one thing though - if I have two different tests that are saving to the same snapshot file, I am getting a permission error on watch: `EPERM: operation not permitted, lstat`.

**Comment by thomasbertet** [comment]
@igor-dv I'm thinking this is not related to this fix, is it ? 
I'm not really sure how this change could do anything on the file system :zipper_mouth_face:

**Comment by igor-dv** [comment]
I think that the problem is that when adding a watch flag, jest starts watching changes on files, and if two different tests are editing the same snapshot (the same file), it conflicts because this snapshot file is locked or something like this. I think this scenario is rare, so I will approve the changes to check them with storyshots.

**Review by igor-dv — APPROVED** [review]

**Pull request merged**